types/react-helmet/v4/index.d.ts

```diff
@@ -8,7 +8,7 @@
 
 import * as React from "react";
 
-declare function ReactHelmet(): ReactHelmet.HelmetComponent;
+declare class ReactHelmet extends React.Component<ReactHelmet.HelmetProps> {}
```

Comment:
Same here; the code is confusing, but helmet defines a factory function and then exports _its result_, not the factory function itself.